Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -530,14 +571,19 @@ public struct MaxKey: BSONValue, Equatable, Codable {
 
     public func encode(to storage: DocumentStorage, forKey key: String) throws {
         guard bson_append_maxkey(storage.pointer, key, Int32(key.count)) else {
-            throw bsonEncodeError(value: self, forKey: key)
+            throw bsonTooLargeError(value: self, forKey: key)
         }
     }
 
     /// Initializes a new `MaxKey` instance.
     public init() {}
 
-    public static func from(iterator iter: DocumentIterator) -> MaxKey { return self.init() }
+    public static func from(iterator iter: DocumentIterator) throws -> MaxKey {
+        guard iter.currentType == .maxKey else {
+            throw wrongIterTypeError(iter, expected: MaxKey.self)
+        }
+        return self.init()
```

Comment:
to be consistent with `BSONNull`'s `from` method, maybe we should we change `self.init()` to `MaxKey()`? (same q for `MinKey`)